### PR TITLE
Remove csv parsing

### PR DESCRIPTION
**Pull request by shaharyarn**

### CODE REVIEW

File: marine/marine.py [hunk 1]
```diff
@@ -186,11 +184,12 @@ def validate_fields(
...
 
     @staticmethod
-    def _parse_output(output: str) -> List[Optional[str]]:
+    def _parse_output(output: List[str], length: int) -> List[Optional[str]]:
         # TODO: this is a bottleneck. Find a better way to provide output from the c code
-        f = StringIO(output)
-        csv_parsed_output = next(csv.reader(f, delimiter="\t", quotechar='"'), [""])
-        return [value or None for value in csv_parsed_output]
+        return (
+            output[i].decode("utf-8") if output[i] is not None else None
```
Comment (yehonatanz) [inline marine/marine.py]: Can `output[i]` actually be `None`? If so, change the typing.
Also, is `output` an actual `list` or is it some `ctypes` array?

**Review by yehonatanz — APPROVED** [review]

File: marine/marine.py (hunk 1 above)
Comment (yehonatanz) [inline marine/marine.py]: Also `str` has no `decode` in python3, so that should probably be `bytes` anyway

**Review by yehonatanz — DISMISSED** [review]

File: marine/marine.py [hunk 2]
```diff
@@ -186,11 +184,12 @@ def validate_fields(
         return bool(self._marine.validate_fields(fields_c_arr, fields_len))
 
     @staticmethod
-    def _parse_output(output: str) -> List[Optional[str]]:
+    def _parse_output(output: List[str], length: int) -> List[Optional[str]]:
         # TODO: this is a bottleneck. Find a better way to provide output from the c code
```
Comment (tomlegkov) [inline marine/marine.py]: Is this still a bottleneck? How much faster is this code now?

**Review by tomlegkov — APPROVED** [review]

File: marine/marine.py (hunk 1 above)
Comment (domerd) [reply marine/marine.py]: Yes, `output[i]` can be None.
`output` is not a list but rather `POINTER(c_char_p)` which is more like a `List[bytes]` like you mentioned in the next comment

Comment (domerd) [reply marine/marine.py]: Discussed above

File: marine/marine.py (hunk 2 above)
Comment (domerd) [reply marine/marine.py]: On average the code is 5% faster.
In some cases (in tests without fields) we slowed the test a bit - might be some overhead we are paying for.

**Pull request merged**